core/src/main/java/org/apache/iceberg/rest/RESTSessionCatalog.java

```diff
@@ -202,8 +200,12 @@ public void initialize(String name, Map<String, String> unresolved) {
 
     this.reporter = CatalogUtil.loadMetricsReporter(mergedProps);
 
-    this.reportingViaRestEnabled =
-        PropertyUtil.propertyAsBoolean(mergedProps, REST_METRICS_REPORTING_ENABLED, true);
+    if (PropertyUtil.propertyAsBoolean(mergedProps, REST_METRICS_REPORTING_ENABLED, true)) {
+      this.reporter =
+          MetricsReporters.combine(
+              this.reporter, new RESTMetricsReporter(client, paths, catalogAuth.headers()));
```

Comment:
Style: I think it would be better to avoid all of the complex formatting by moving `new RESTMetricsReporter(...)` to its own statement.